core/src/io/anuke/mindustry/world/BlockStorage.java

```diff
@@ -143,7 +144,7 @@ public float tryMoveLiquid(Tile tile, Tile next, boolean leak, Liquid liquid){
 
             if(next.block().acceptLiquid(next, tile, liquid, 0f)){
                 float ofract = next.entity.liquids.get(liquid) / next.block().liquidCapacity;
-                float fract = tile.entity.liquids.get(liquid) / liquidCapacity;
+                float fract = tile.entity.liquids.get(liquid) / liquidCapacity * tile.block().liquidPressure;
```

Comment:
This can just be `liquidPressure`, as you are already calling this as the transportation block.